src/ImageSharp/Formats/Tiff/Ifd/DirectoryReader.cs

```diff
@@ -49,8 +49,8 @@ public IEnumerable<ExifProfile> Read()
 
         private static ByteOrder ReadByteOrder(Stream stream)
         {
-            byte[] headerBytes = new byte[2];
-            stream.Read(headerBytes, 0, 2);
+            Span<byte> headerBytes = stackalloc byte[2];
```

Comment:
@JimBobSquarePants we use a lot of `stackalloc`s for local buffers. By default the JIT must emit code to zero the buffer contents. `[SkipLocalsInit]` tells the JIT that this zeroing shouldn't be done, hence the produced machinecode is more compact, and yields better perf. This attribute can be applied to individual methods, types, modules, ...
I'd like to enable it globally (i.e. per module), so that all occurences of stackalloc will benefit from this.

Is there any policy for this? In regards of scratch buffers not being zeroed?

For older targets, where this attribute isn't available, one can create a dummy attribute with same typename, to make it work.

Reply:
We’ve been very careful to ensure we don’t use what we don’t write so we should be fine to enable it.